lib/posix/posix_internal.h

```diff
@@ -31,10 +31,10 @@ enum pthread_state {
 	PTHREAD_DETACHED = PTHREAD_CREATE_DETACHED,
 	/* The thread is running and joinable. */
 	PTHREAD_JOINABLE = PTHREAD_CREATE_JOINABLE,
+	/* A joinable thread exited and its return code is available. */
+	PTHREAD_EXITED,
 	/* The thread structure is unallocated and available for reuse. */
```

Comment:
This name doesn't match the described semantic very well; is this more like `PTHREAD_NOT_IN_USE` or `PTHREAD_IDLE`?

Reply:
Agreed. There is a distinct ambiguity between `TERMINATED` and `EXITED`. For the time being, I'd prefer to keep naming though, to ensure that backports are consistent.